### PR TITLE
add CI evals

### PR DESCRIPTION
**Pull request by vbarda**

### CODE REVIEW

**Comment by vbarda** [comment]
cc @rlancemartin

File: backend/tests/evals/test_e2e.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+DATASET_NAME = "chat-langchain-qa"
+EXPERIMENT_PREFIX = "chat-langchain-ci"
+
+SCORE_RETRIEVAL_RECALL = "retrieval_recall"
+SCORE_ANSWER_CORRECTNESS = "answer_correctness_score"
+SCORE_ANSWER_VS_CONTEXT_CORRETNESS = "answer_vs_context_correctness_score"
+
+# claude sonnet / gpt-4o are a bit too expensive
+JUDGE_MODEL_NAME = "claude-3-haiku-20240307"
```
Comment (vbarda) [inline backend/tests/evals/test_e2e.py:22]: ideally of course it's something like 3.5 sonnet, but it's a bit pricey at ~$1 or so per eval run

File: backend/tests/evals/test_e2e.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+class GradeAnswer(BaseModel):
+    """Continuous score to assess the correctness of the answer."""
+
+    score: float = Field(
+        description="How correct is the answer? 1.0 if completely correct or 0.0 if completely incorrect",
+        minimum=0.0,
+        maximum=1.0,
+    )
+    reason: str = Field(
```
Comment (hinthornw) [inline backend/tests/evals/test_e2e.py:51]: I'd put reason first

File: backend/tests/evals/test_e2e.py (hunk 1 above)
Comment (hinthornw) [reply backend/tests/evals/test_e2e.py:22]: Ya haiku should be fine - can check correlation and self-improve if we want

Comment (hinthornw) [reply backend/tests/evals/test_e2e.py:22]: We could also just configure these in-app on the dataset and add some self-improvement

File: backend/tests/evals/test_e2e.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+from langsmith.schemas import Example, Run
+
+from backend.graph import OPENAI_MODEL_KEY, format_docs, graph
+
+DATASET_NAME = "chat-langchain-qa"
+EXPERIMENT_PREFIX = "chat-langchain-ci"
+
+SCORE_RETRIEVAL_RECALL = "retrieval_recall"
+SCORE_ANSWER_CORRECTNESS = "answer_correctness_score"
+SCORE_ANSWER_VS_CONTEXT_CORRETNESS = "answer_vs_context_correctness_score"
```
Comment (hinthornw) [inline backend/tests/evals/test_e2e.py:19]: nit: spelling of variable

**Review by hinthornw — APPROVED** [review]
Seems reasonable. Got a link to the results?

File: backend/tests/evals/test_e2e.py (hunk 2 above)
Comment (vbarda) [reply backend/tests/evals/test_e2e.py:51]: sure, ooc is this empirically better?

**Comment by vbarda** [comment]
> Seems reasonable. Got a link to the results?

https://smith.langchain.com/o/ebbaf2eb-769b-4505-aca2-d11de10372a4/datasets/1adfbbb1-8039-49a9-b634-ca1393f2ccd0?paginationState=%7B%22pageIndex%22%3A0%2C%22pageSize%22%3A10%7D

File: .github/workflows/eval.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+      - name: Install dependencies
+        run: poetry install --with dev
+
+      - name: Evaluate
+        env:
+          LANGCHAIN_API_KEY: ${{ secrets.LANGCHAIN_API_KEY }}
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          ANTHROPIC_API_KEY: ${{ secrets.ANTHROPIC_API_KEY }}
+          WEAVIATE_URL: ${{ secrets.WEAVIATE_URL }}
```
Comment (hinthornw) [inline .github/workflows/eval.yml:39]: At some point we could see if caching works well here. I'd kinda assume the docs change enough that it wouldn't

**Pull request merged**